### ISSUE TITLE
Predicated Rendering not working inside renderdoc

### ISSUE THREAD

**Issue by redorav** [baldurk/renderdoc#2085]
## Description

I have set up predicated rendering in an application, I can verify visually that it works, but a renderdoc capture will show it rendering regardless of the state of the predication. I can see the calls to Begin/End and SetPredication, but they aren't respected (i.e. objects are rendered regardless)

I am using predicated rendering to cull shadow rendering if a light is not visible. I can verify it working the other way around (i.e. do not render shadows if light is visible). My expectation would be that renderdoc does not render the shadow drawcalls.

## Steps to reproduce

Basically as far as I can tell any use of predicated rendering would be broken as this is a really simple case. I cannot really send an application but I can send a capture if that helps.

## Environment

* RenderDoc version: 1.10
* Operating System: Windows 10
* Graphics API: D3D11
* GPU: RTX 2060

**Comment by baldurk** [comment]
Yes RenderDoc deliberately doesn't replay predication since it's not generally possible to replay it reliably and it behaves inconsistently when applied.

The predication state is displayed in the rasterizer state tab in the pipeline viewer showing whether it would have passed or failed, but that can include false negatives (i.e. showing it as passing when it would have failed).

**Comment by redorav** [comment]
Could you give a bit more detail as to why it's not reliable? It's more out of curiosity at this point, I trust that you've thought about this problem before.

**Comment by baldurk** [comment]
This came up quite a few years ago so my memory is a little vague but I believe a large part of the problem is predicate results being used but unavailable from previous frames prior to the capture. You can end up in a situation where sometimes predication works and sometimes it doesn't seemingly at random to the user, which can cause you to start investigating something that isn't actually a bug at all. Predication is already easy to miss as a cause of missing drawcalls so I didn't want to make it worse by adding unreliable element. It also doesn't really jibe with the rest of the UI which act like the drawcall is happening (including e.g. the texture overlays).

AFAIK all APIs treat predicated/conditional rendering as a conservative optimisation hint rather than mandating results, so there shouldn't be a correctness issue in rendering the draws still. Like I say the results are available in the rasterizer state where possible so you can still introspect what's happening in the case where the results are in-frame.

**Comment by redorav** [comment]
Excellent, thanks for taking the time to explain, it makes perfect sense. I didn't know the result was available in the rasterizer but it's most helpful and I can now see what I wanted. I wonder if there's a way to maybe determine whether a query is actually updated within the frame or not and cater for that specific use case, it's a very low priority task anyway. Thanks!
